course/layouts/partials/video.html

```diff
@@ -2,6 +2,9 @@
 {{ $captionsLocation := .Params.video_files.video_captions_file }}
 {{ $transcriptPdfLocation := .Params.video_files.video_transcript_file }}
 {{ $downloadLink := .Params.file }}
+{{ $relatedResourses := .Params.related_resources_text }}
```

Comment:
I wonder why there's no need to give a default value here as the code does not break if this variable does not exist. Maybe these variables are defined in the `front matter`?

Reply:
- The reason we cannot loop over them is mainly both of them behave in somewhat different  ways since related resources has a fixed title but a dynamic content while optional resources have a dynamic title and dynamic content.
The better way to remove duplication would be to create a partial which takes this values as parameters and call them here.
-  These variables are defined in front matter and  are somewhat guaranteed, if not then we would have to add the check for every variable used here

Reply:
- Both, dynamic and fixed content can be handled with array and loop but creating a partial is a more relevant and better approach here I guess.
- Yeah, they exist in the front matter but even if a variable does not exist `(e.g: .Params.anything_here)`, the code still does not break. I wonder why? 😄